packages/edit-site/src/components/save-button/index.js

```diff
@@ -52,12 +54,47 @@ export default function SaveButton( {
 				previewingThemeName: previewingTheme?.name?.rendered,
 			};
 		}, [] );
+
 	const { setIsSaveViewOpened } = useDispatch( editSiteStore );
+	const { customizedSaveButtonAction, customizedSaveButtonLabel } = useSelect(
+		( select ) => {
+			const { getSettings } = unlock( select( editSiteStore ) );
+			return {
+				customizedSaveButtonAction:
+					getSettings().__experimentalSaveButtonAction,
+				customizedSaveButtonLabel:
+					getSettings().__experimentalSaveButtonLabel,
+			};
+		},
+		[]
+	);
+	const onClick = useCallback( () => {
+		if ( customizedSaveButtonAction ) {
+			customizedSaveButtonAction();
+			return;
+		}
+		setIsSaveViewOpened( true );
+	}, [ customizedSaveButtonAction, setIsSaveViewOpened ] );
+
+	// For testing
+	const { updateSettings } = useDispatch( editSiteStore );
+	useEffect( () => {
+		updateSettings( {
+			__experimentalSaveButtonAction: () =>
+				// eslint-disable-next-line no-console
+				console.log( 'Customized Action' ),
+			__experimentalSaveButtonLabel: 'Customized Action',
+		} );
+	}, [ updateSettings ] );
```

Comment:
I'll revert this diff after testing. https://github.com/WordPress/gutenberg/pull/56807/commits/9929ca0760f3ff334939e5c08ec314092124318c

Reply:
Don't forget to do this!